cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -275,6 +276,18 @@ bool CompactProtocolReader::read(DictionaryPageHeader* d)
   return function_builder(this, op);
 }
 
+bool CompactProtocolReader::read(DataPageHeaderV2* d)
```

Comment:
How come this wasn't needed in https://github.com/rapidsai/cudf/pull/11778?

Reply:
Because the header decoder is device code, so there's a duplicate thrift decoder buried in `page_hdr.cu`  We've got little bits of thrift decoding sprinkled all over the place :(